docs/content/v0.20/architecture.md

```diff
@@ -444,3 +484,25 @@ of tracked objects in the namespace associated with the WEC cluster.
 A `WorkStatus` object contains status for exactly one object, so that 
 status updates for one object do not require updates of a whole bundle. 
 
+
+### Transport Controller - PLANNED TO BE ADDED
+
+The transport controller is pluggable and allows the option to plug different
+implementations of the transport interface. In order to implement transport plugin, the plugin should supply the following:
+- upon registration of a new WEC, plugin should create a namespace per WEC in ITS (mailbox namespace per WEC);
+- plugin must be able to wrap multiple objects into a single wrapped object;
+- have an agent that runs on the WEC and watches the wrapped object in the corresponding namespace in the central hub and is able to unwrap it and apply the objects to the WEC;
```

Comment:
(A) any number of wrapped objects, not necessarily one.

(B) "an agent that runs on  the WEC and watches the wrapped objects in the corresponding namespace" is overly specific. Much more indirect or complicated techniques should be allowed.

Reply:
(A) I changed but not exactly as you described it. I wrote:
plugin must be able to wrap any number of objects into a single wrapped object. once the plugin can wrap any number of objects, it can wrap individual object to a wrapped object multiple times (e.g., each individual object gets its own wrapped object), or wrap multiple objects to a single wrapped object. 

(B) fixed

Reply:
Since writing my original comment here I have realized that you are documenting a different design than I thought. Let's pursue the debate about what the design should be separately.

If I understand correctly, the design you are trying to document has an additional interface: the interface between the generic transport controller code and the plugin code. The interface to the plugin is NOT the kube API objects, it is a golang interface that the generic transport controller calls. Have I got this right? If so then it should be explicitly discussed in this document. If not then I am still puzzled by the language that I see.

Reply:
I've added 2 more sentences describing the go interface. I don't want to go too much into the details, I think it should be described at the high level. I've added a link to `transport.go` interface and to `ocm-transport-plugin` repo, so if someone wants more details he can go to the links.